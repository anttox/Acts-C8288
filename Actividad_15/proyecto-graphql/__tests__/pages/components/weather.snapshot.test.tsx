/**
 * @jest-environment jsdom
 */

import { act, create } from "react-test-renderer";
import PageComponentWeather from "../../../pages/components/weather";

describe("PageComponentWeather", () => {
    test("renders correctly", () => {
        const component = create(
            <PageComponentWeather
                zipCode="15551"
                weather="Sunny"
                tempC="28"
                tempF="82.4"
                friends={["15552", "15553"]}
            />
        );

        // Verifica que el componente no sea null y procesa la instantánea
        const tree = component.toJSON();
        if (!tree) {
            throw new Error("Component failed to render");
        }
        expect(tree).toMatchSnapshot();
    });

    test("clicks the h1 element and updates the state", async () => {
        let component: any;

        await act(async () => {
            component = create(
                <PageComponentWeather
                    zipCode="15551"
                    weather="Sunny"
                    tempC="28"
                    tempF="82.4"
                    friends={["15552", "15553"]}
                />
            );
        });

        // Asegúrate de que el componente esté montado antes de buscar el h1
        if (!component) {
            throw new Error("Component was not mounted");
        }

        const heading = component.root.findByType("h1");

        // Simula el clic en el encabezado (h1)
        await act(() => {
            heading.props.onClick();
        });

        // Verifica la instantánea después del clic
        const tree = component.toJSON();
        if (!tree) {
            throw new Error("Component failed to render after click");
        }
        expect(tree).toMatchSnapshot();
    });
});
