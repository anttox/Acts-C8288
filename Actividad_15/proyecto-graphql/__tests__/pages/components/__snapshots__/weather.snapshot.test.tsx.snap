// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageComponentWeather clicks the h1 element and updates the state 1`] = `
<div>
  <h1
    data-testid="weather-heading"
    onClick={[Function]}
  >
    Weather Information
  </h1>
  <p
    data-testid="zip-code"
  >
    Zip Code: 
    15551
  </p>
  <p
    data-testid="weather-status"
  >
    Weather: 
    Sunny
  </p>
  <p
    data-testid="temperature-c"
  >
    Temperature: 
    28
    째C / 
    82.4
    째F
  </p>
  <p
    data-testid="friends"
  >
    Nearby Friends' Zip Codes: 
    15552, 15553
  </p>
  <p
    data-testid="counter"
  >
    Counter: 
    1
  </p>
</div>
`;

exports[`PageComponentWeather renders correctly 1`] = `
<div>
  <h1
    data-testid="weather-heading"
    onClick={[Function]}
  >
    Weather Information
  </h1>
  <p
    data-testid="zip-code"
  >
    Zip Code: 
    15551
  </p>
  <p
    data-testid="weather-status"
  >
    Weather: 
    Sunny
  </p>
  <p
    data-testid="temperature-c"
  >
    Temperature: 
    28
    째C / 
    82.4
    째F
  </p>
  <p
    data-testid="friends"
  >
    Nearby Friends' Zip Codes: 
    15552, 15553
  </p>
  <p
    data-testid="counter"
  >
    Counter: 
    0
  </p>
</div>
`;
